### PR TITLE
sync Dockerfile.in from downstream

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* syncs the Dockerfile.in from downstream for consistency

cc @Red-GV @alanconway

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,33 +9,22 @@ ENV OS_GIT_PATCH=${CI_Z_VERSION}
...
 RUN make build
 
-
-#@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-cli:v4.8)
+#@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-cli:v4.9)
 FROM quay.io/openshift/origin-cli:4.8 AS origincli
 
 #@follow_tag(registry.redhat.io/ubi8:latest)
 FROM registry.redhat.io/ubi8:8.5
+ENV REMOTE_SOURCE_DIR=${REMOTE_SOURCE_DIR:-/remote_source}
```
Comment (Red-GV) [inline Dockerfile:27]: We actually use `REMOTE_SOURCES` instead of `REMOTE_SOURCE_DIR`.

Comment (jcantrill) [reply Dockerfile:27]: This is the same as we do for MOLO and as advised in CPaaS users group. Ref:
* https://source.redhat.com/groups/public/container-build-system/container_build_system_wiki/containers_from_source_multistage_builds_in_osbs#jive_content_id_Cachito_Integration_for_gomod
* https://chat.google.com/room/AAAATiT7PR8/CqYbPOM1Ip8

This seems wrong and I'm surprised it works https://gitlab.cee.redhat.com/openshift-logging/cpaas-config/-/blob/openshift-logging-5.4-rhel-8/distgit/containers/cluster-logging-operator/Dockerfile.in#L11

Comment (jcantrill) [reply Dockerfile:27]: I'll update per our conversation with team

**Pull request merged**